### PR TITLE
Refactor movies and cinema halls into monte way

### PR DESCRIPTION
**Pull request by atytko**
Refactored movies and cinema halls into MonteWay.

### CODE REVIEW

File: app/concepts/cinema_halls/repository.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    def update(id, params)
+      cinema_hall = adapter.find(id)
+      cinema_hall.update(params)
+      cinema_hall
+    end
+
+    def delete(id)
+      adapter.find(id).delete
+    end
```
Comment (wmaciejak) [inline app/concepts/cinema_halls/repository.rb:31]: Now, when you have similar behavior for more than one repository, you can try to create BaseRepository class. Of course it should be moved to next PR and consider it like an additional thing to do :)

File: app/controllers/cinema_halls_controller.rb [hunk 2]
```diff
@@ -5,40 +5,37 @@ class CinemaHallsController < ApplicationController
...
       render jsonapi: cinema_hall.errors, status: :unprocessable_entity
     end
   end
 
   def update
-    cinema_hall = CinemaHall.find(params[:id])
+    cinema_hall = CinemaHalls::UseCases::Update.new.call(id: params[:id], params: cinema_hall_params)
     authorize cinema_hall
-    if cinema_hall.update(cinema_hall_params)
+    if cinema_hall.valid?
```
Comment (wmaciejak) [inline app/controllers/cinema_halls_controller.rb:30]: Consider that if somebody will not have access to this particular cinema_hall, he still will be able to update this object, but he will get only unauthorized response.

File: app/controllers/cinema_halls_controller.rb [hunk 3]
```diff
@@ -5,40 +5,37 @@ class CinemaHallsController < ApplicationController
...
       render jsonapi: cinema_hall
     else
       render jsonapi: cinema_hall.errors, status: :unprocessable_entity
     end
   end
 
   def destroy
-    cinema_hall = CinemaHall.find(params[:id])
+    cinema_hall = CinemaHalls::UseCases::Delete.new.call(id: params[:id])
```
Comment (wmaciejak) [inline app/controllers/cinema_halls_controller.rb:38]: Here is the same problem - even if you don't have access to object, you can remove it.

File: app/controllers/cinema_halls_controller.rb [hunk 4]
```diff
@@ -5,40 +5,37 @@ class CinemaHallsController < ApplicationController
...
+    cinema_hall = CinemaHalls::UseCases::FetchOne.new.call(id: params[:id])
     authorize cinema_hall
-
     render jsonapi: cinema_hall, except: blacklisted_attributes
   end
 
   def create
-    cinema_hall = CinemaHall.new(cinema_hall_params)
+    cinema_hall = CinemaHalls::UseCases::Create.new.call(params: cinema_hall_params)
     authorize cinema_hall
```
Comment (wmaciejak) [inline app/controllers/cinema_halls_controller.rb:19]: And here the same

File: app/controllers/cinema_halls_controller.rb (hunk 2 above)
Comment (atytko) [reply app/controllers/cinema_halls_controller.rb]: Thanks for spotting this, corrected :)

File: app/controllers/cinema_halls_controller.rb [hunk 5]
```diff
@@ -5,42 +5,38 @@ class CinemaHallsController < ApplicationController
...
       render jsonapi: cinema_hall.errors, status: :unprocessable_entity
     end
   end
 
   def update
-    cinema_hall = CinemaHall.find(params[:id])
-    authorize cinema_hall
-    if cinema_hall.update(cinema_hall_params)
+    authorize CinemaHall
+    cinema_hall = CinemaHalls::UseCases::Update.new.call(id: params[:id], params: cinema_hall_params)
```
Comment (wmaciejak) [inline app/controllers/cinema_halls_controller.rb:29]: Here you can consider that first of all you can try to find this exact record, validate authorization and then update it. 

So consider this approach:

```ruby
  def update
    cinema_hall =  CinemaHalls::UseCases::FetchOne.new.call(id: params[:id])
    authorize cinema_hall
    
    updated_cinema_hall = CinemaHalls::UseCases::Update.new.call(id: params[:id], params: cinema_hall_params)
    ...
  end
```

File: app/controllers/movies_controller.rb [hunk 6]
```diff
@@ -5,41 +5,36 @@ class MoviesController < ApplicationController
...
     end
   end
 
   def update
-    movie = Movie.find(params[:id])
-    authorize movie
-    if movie.update(movie_params)
+    authorize Movie
+    movie = Movies::UseCases::Update.new.call(id: params[:id], params: movie_params)
+    if movie.valid?
```
Comment (wmaciejak) [inline app/controllers/movies_controller.rb:28]: Consider the same approach like mentioned above

**Review by wmaciejak — APPROVED** [review]

**Pull request merged**